include/cutlass/gemm/collective/xe_mma_mixed_input.hpp

```diff
@@ -295,57 +296,58 @@ struct CollectiveMma<
       }
     }();
 
-
+    auto mA_mkl =
+        make_tensor(make_gmem_ptr(args.ptr_A), make_layout(make_shape(M, K, L), args.dA));
     auto mB_nkl =
         make_tensor(ptr_B, make_layout(make_shape(N, K, L), args.dB));
 
     Copy_A tiled_copy_a{Copy_A{}.with(mA_mkl)};
     Copy_B tiled_copy_b{Copy_B{}.with(mB_nkl)};
 
-    if constexpr(KernelConversionMode == ConversionMode::DirectConvert){
+    if constexpr(ModeDirectConvert){
       return Params{tiled_copy_a, tiled_copy_b, {}, {}, 0};
-    }
-
-    auto scale_k = cute::ceil_div(K, args.group_size);
-    auto mScale = make_tensor(
-        make_gmem_ptr(static_cast<NonVoidElementScale const *>(args.ptr_S)),
-        make_layout(make_shape(IsATransformed ? M : N, scale_k, L), args.dS));
-    auto tiled_copy_scale = [&]() {
-      if constexpr(quant_mode == QuantMode::GroupWise) {
-        return Copy_Scale{}.with(mScale);
-      } else {
-        return make_tensor(
-          make_gmem_ptr(static_cast<NonVoidElementScale const *>(args.ptr_S)),
-          make_layout(make_shape(_1{}, 1, 1), args.dS));
-      }
-    }();
-
-    if constexpr(KernelConversionMode == ConversionMode::ConvertAndScale){
-      return Params{tiled_copy_a, tiled_copy_b, tiled_copy_scale, {}, args.group_size};
-    }
+    } else if constexpr (ModeScale || ModeScaleZero) {
+      auto tiled_copy_scale = [&]() {
+        if constexpr(is_groupwise) {
+          auto scale_k = cute::ceil_div(K, args.group_size);
+          auto mScale = make_tensor(make_gmem_ptr(static_cast<NonVoidElementScale const *>(args.ptr_S)),
+                                    make_layout(make_shape(IsATransformed ? M : N, scale_k, L), args.dS));
+          return Copy_Scale{}.with(mScale);
+        } else {
+          return make_tensor(
+            make_gmem_ptr(static_cast<NonVoidElementScale const *>(args.ptr_S)),
+            make_layout(make_shape(_1{}, 1, 1), args.dS));
+        }
+      }();
 
-    auto ptr_Z = [&]() {
-      if constexpr (sizeof_bits_v<NonVoidElementZero> < 8) {
-        return cute::subbyte_iterator<const NonVoidElementZero>(args.ptr_Z);
+      if constexpr (ModeScale) {
+        return Params{tiled_copy_a, tiled_copy_b, tiled_copy_scale, {}, args.group_size};
       } else {
-        return make_gmem_ptr(static_cast<NonVoidElementZero const *>(args.ptr_Z));
-      }
-    }();
+        auto ptr_Z = [&]() {
+          if constexpr (sizeof_bits_v<NonVoidElementZero> < 8) {
+            return cute::subbyte_iterator<const NonVoidElementZero>(args.ptr_Z);
+          } else {
+            return make_gmem_ptr(static_cast<NonVoidElementZero const *>(args.ptr_Z));
+          }
+        }();
+
+        auto tiled_copy_zero = [&](){
```

Comment:
Capturing buinding structure is c++20 feature, we are using c++-17, So it cause warning in the code.
```suggestion
        auto tiled_copy_zero = [&](){
        auto [M, N, K, L] = problem_shape;

```